pageserver/page_api/src/model.rs

```diff
@@ -102,6 +102,15 @@ impl TryFrom<ReadLsn> for proto::ReadLsn {
     }
 }
 
+impl From<&ReadLsn> for proto::ReadLsn {
```

Comment:
Let's drop these `From<&>` implementations. We should generally only convert owned types to avoid unnecessary heap allocations, and the caller can copy or clone if necessary.